### DISCUSSION TITLE
Will this package work after 31/03/2023? The Google Sign-In JavaScript platform library for Web is deprecated

### DISCUSSION THREAD

**Discussion by watjuaninacio** [abacritt/angularx-social-login#648] [General]
Hi there, I got an email last Septembre about the Google Sign-In JavaScript platform library for Web being deprecated. I'm currently using this package for google sign in, will I need to change anything in the code in order for my app to continue working after March 31st on 2023? Thanks!

(https://developers.googleblog.com/2021/08/gsi-jsweb-deprecation.html), and unavailable for download after March 31, 2023. The solutions in this guide are based on this library and therefore also deprecated.

Use instead the new [Google Identity Services for Web](https://developers.google.com/identity/gsi/web/guides/overview) solution to quickly and easily sign users into your app using their Google accounts.

By default, new client IDs are now blocked from using the older platform library; existing client IDs are unaffected. New client IDs created before July 29th, 2022 may set the [plugin_name](https://developers.google.com/identity/sign-in/web/reference#gapiauth2clientconfig) to enable use of the legacy Google platform library.

https://developers.google.com/identity/gsi/web/guides/migration#rendered

**Comment by nwalkewicz** [comment]
Hey there,

I believe this package is already using the new Sign In With Google client library, so you won't have to change anything in your code due to that deprecation. Notice a couple key features for the new client library in the [deprecation notice](https://developers.googleblog.com/2021/08/gsi-jsweb-deprecation.html) such as personalized buttons and the One Tap prompt. By default, this package uses the One Tap prompt. If you choose to set `<asl-google-signin-button>`'s size to `large`, it'll automatically show it as "personalized". (Refer to the [docs](https://github.com/abacritt/angularx-social-login#sign-in-with-google) for Google Signin button usage.) These features wouldn't be possible with the old client library.

**Reply by santacruzfreddy** [reply]
Hello, this is for all version of angularx-social-login or only lastest version, I asked  because I used angularx-social-login@1.2.7 and I don't know If this version has problem with update google api.

**Reply by ReinierMP** [reply]
Hi @santacruzfreddy, finally is this issue fixed for all versions or just the latest?

**Reply by santacruzfreddy** [reply]
I had to update my project to angular 15, because my project is only for to login.

**Reply by nwalkewicz** [reply]
@santacruzfreddy Sorry, I didn't see your last reply. I'm currently on angular 14 and the library seems to be working fine for me.

**Reply by nwalkewicz** [reply]
Also, I'm using v1.2.5 of this package if that info helps at all.

**Reply by ReinierMP** [reply]
I am using v3.5.7 for Angular 9 and is still using the https://apis.google.com/js/platform.js and not the https://accounts.google.com/gsi/client which is the new library to use.

**Reply by DamianRuczynski** [reply]
Hello all,
Quick question: Is Angularx-Social-Login (version 3) still supported for Angular 10? Also, does it handle migrating from the Google Login API's JavaScript implementation?
Thanks in advance for any info!

**Comment by dev-telescopic** [comment]
Hi @nwalkewicz , thanks for clarifying that !
